build.gradle

```diff
@@ -1,23 +1,37 @@
-allprojects {
-  buildscript {
-    repositories {
-      mavenCentral()
-      google()
-    }
+buildscript {
+  repositories {
+    mavenCentral()
+    google()
   }
   dependencies {
-    repositories {
-      mavenCentral()
-      google()
-    }
+    classpath libs.androidGradlePlugin
+    classpath libs.maven.publish.gradlePlugin
   }
+}
 
-  ext {
-    compileSdkVersion = 30
-    targetSdkVersion = 29
-    minSdkVersion = 14
+allprojects {
+  repositories {
+    mavenCentral()
+    google()
   }
+}
+
+subprojects {
+  plugins.withId('com.android.base') {
+    android {
+      defaultConfig {
+        minSdk 14
+        compileSdk 30
+        targetSdkVersion 29
+
+        group GROUP
+        version VERSION_NAME
```

Comment:
These are properties of the Project and should be outside the plugin config. They're not specific to Android.